File: test_upload.py
import requests

url = "http://localhost:3002/api/device/voice"
headers = {
    "X-Device-ID": "28:84:85:42:1C:74",
    "X-Device-Timestamp": "1672531200",
    "X-Device-Signature": "dummy" # if secret is off
}
files = {
    "audio": ("record.wav", b"RIFF\x24\x00\x00\x00WAVEfmt ", "audio/wav")
}

response = requests.post(url, headers=headers, files=files)
print(response.status_code)
print(response.text)

File: test_upload.js
const http = require('http');

const options = {
  hostname: 'localhost',
  port: 3002,
  path: '/api/device/voice',
  method: 'POST',
  headers: {
    'Content-Type': 'audio/wav',
    'X-Device-ID': 'test_mac'
  }
};

const req = http.request(options, (res) => {
  console.log(`STATUS: ${res.statusCode}`);
  res.setEncoding('utf8');
  res.on('data', (chunk) => {
    console.log(`BODY: ${chunk}`);
  });
});

req.on('error', (e) => {
  console.error(`problem with request: ${e.message}`);
});

// Write a fake WAV header and some data
const fakeWav = Buffer.alloc(100);
req.write(fakeWav);
req.end();
